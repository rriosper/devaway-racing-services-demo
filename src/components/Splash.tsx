import styled, { css } from '@xstyled/styled-components';
import { up } from '@xstyled/system';
import { motion } from 'framer-motion';
import React, { useState } from 'react';

const SSplash = styled(motion.div)`
  position: fixed;
  top: 0;
  left: 0;
  z-index: splash;
  width: 100%;
  height: 100%;
  background: white;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const SImage = styled(motion.img)`
  width: 80px;
  height: 80px;

  ${up(
    'sm',
    css`
      width: 150px;
      height: 150px;
    `,
  )}
`;

const Splash: React.FC = () => {
  const [isActive, setIsActive] = useState(true);
  return isActive ? (
    <SSplash
      animate={{
        opacity: [1, 1, 1, 0],
      }}
      transition={{ duration: 1.4, ease: 'easeOut' }}
      onAnimationComplete={() => setIsActive(false)}
    >
      <SImage
        alt="DevAway logo"
        src="/images/devAway.jpeg"
        animate={{
          opacity: [0, 1, 1],
          boxShadow: ['0 0 20px rgba(0,0,0,0.5)', '0 0 20px rgba(0,0,0,0)'],
          transform: ['scale(1.1)', 'scale(1)'],
        }}
        transition={{ duration: 0.4, ease: 'easeIn' }}
      />
    </SSplash>
  ) : null;
};

export default Splash;
