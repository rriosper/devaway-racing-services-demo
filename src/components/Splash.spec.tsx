import React from 'react';
import { render } from '@testing-library/react';

import Splash from './Splash';

test('Splash', () => {
  const { asFragment } = render(<Splash />);
  expect(asFragment).toMatchSnapshot();
});
