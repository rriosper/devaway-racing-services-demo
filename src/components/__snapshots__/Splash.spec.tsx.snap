// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash 1`] = `[Function]`;
